baseband_tasks/dispersion.py

```diff
@@ -182,3 +182,59 @@ def __init__(self, ih, dm, reference_frequency=None,
     @property
     def dm(self):
         return -self._dm
+
+
+class DisperseShift(SampleShift):
+    """Incoherently shift a time stream based on the disperse time delay.
+
+    This task does not handle the in channel smearing, only shift the samples.
+
+    Parameters
+    ----------
+    ih : task or `baseband` stream reader
+        Input data stream, with time as the first axis.
+    dm : float or `~baseband_tasks.dm.DispersionMeasure` quantity
+        Dispersion measure.
+    reference_frequency : `~astropy.units.Quantity`
+        Frequency to which the data should be dedispersed.  Can be an array.
+        By default, the mean frequency.  If one doesn't want to change the
+        start time, choose the maximum frequency.
+    samples_per_frame : int, optional
+        Number of dedispersed samples which should be produced in one go.
+        The number of input samples used will be larger to avoid wrapping.
+        If not given, as produced by the minimum power of 2 of input
+        samples that yields at least 75% efficiency.
+    frequency : `~astropy.units.Quantity`, optional
+        Frequencies for each channel in ``ih`` (channelized frequencies will
+        be calculated).  Default: taken from ``ih`` (if available).
+    sideband : array, optional
+        Whether frequencies in ``ih`` are upper (+1) or lower (-1) sideband.
+        Default: taken from ``ih`` (if available).
+    """
+    def __init__(self, ih, dm, reference_frequency=None,
+                 samples_per_frame=None, frequency=None, sideband=None):
+        # Compute the time shift
+        # TODO: 1. How to remove the duplicated code
+        # 2. Is calculating delay based on the input frequency the right thing
+        # to do?
+        dm = DispersionMeasure(dm)
+        frequency = getattr_if_none(ih, 'frequency', frequency)
+        sideband = getattr_if_none(ih, 'sideband', sideband)
+
+        # Calculate frequencies at the top and bottom of each band.
```

Comment:
As just discussed, let's not worry about getting the reference frequency to this level - let's just default to `reference_frequency = frequency.mean()`.